### DISCUSSION TITLE
Ungrateful

### DISCUSSION THREAD

**Discussion by krischik** [krischik/vim-ada#22] [General]
Hello 

I'm kind of surprised how ungrateful the Ada and or Vim community is. After a few years I updated the Ada Bundle and all I get is users whining about _„missing features“_ or (undocumented) _„standards not followed“_. Not a thank you to be seen. I don't think the community was like this when I started working on the Ada bundle in 2005.

The icing on the cake was a pull request which disabled core functionality. I'm not just talking about key mappings. The commands and menus where also disabled which would have broken the build system. Which makes me believe that there is a misunderstanding on what this plugin is.

This bundle is meant as an **integrated development environment** for those who don't have access to the GNAT Programming Studio. As such it includes menus, commands, key mappings, compiler support, tag and omni completion.

Once an Ada file is opened it takes over your Vim session for Ada development. And that is intentional. It's not meant to be a light weight filetype plugin which unloads itself again. `autoload/ada.vim' has a little over 500 lines. There is no turning back once those are loaded.

Still I will follow established (undocumented) standards and did implement Enhancement #19 will implement Enhancement #21 as well. I do consider removing the insert key mapping as they are probably not needed. I also have to think of a solution for the documentation. I like having the documentation available in the development copy but the installed copy should come without.

Regards

Martin

**Comment by dkearns** [comment]
G'day,
 
> I'm kind of surprised how ungrateful the Ada and or Vim community is. After a few years I updated the Ada Bundle and all I get is users whining about _„missing features“_ or (undocumented) _„standards not followed“_. Not a thank you to be seen. I don't think the community was like this when I started working on the Ada bundle in 2005.

Bug reports and feature suggestions are not personal slights nor do they represent a general lack of gratitude.  Frankly, you seem rather ungrateful to have engaged users trying to help improve the project and I am also "kind of surprised".

If you've found the last couple of issue reports this annoying I'd sincerely consider taking this project down or locking it.   You're also listed as the maintainer in the distributed runtime files so these reports will possibly just move to email but the tone of messages via that channel are often a bit more personable.

> The icing on the cake was a pull request which disabled core functionality. I'm not just talking about key mappings. The commands and menus where also disabled which would have broken the build system. Which makes me believe that there is a misunderstanding on what this plugin is.

What cake?  There is, unfortunately, no cake.

This was clearly offered as a quick fix (even labelled as such) for anyone else stumbling across the bug who came here to report it.  You're not compelled to merge the PR or even review it.  Perhaps it would have been better added as diff in a comment attached to the issue report but it makes little difference.

I find it incomprehensible that it irritated you.  If I was a user inconvenienced by this bug and with no idea how to fix it I'd be grateful that @TamaMcGlinn offered any sort of workaround in the interim.

> This bundle is meant as an **integrated development environment** for those who don't have access to the GNAT Programming Studio. As such it includes menus, commands, key mappings, compiler support, tag and omni completion.
> 
> Once an Ada file is opened it takes over your Vim session for Ada development. And that is intentional. It's not meant to be a light weight filetype plugin which unloads itself again. `autoload/ada.vim' has a little over 500 lines. There is no turning back once those are loaded.

And yet that is exactly how it's distributed in the Vim runtime, as a file type plugin in the more general sense.

This is irrelevant anyway, as one would assume that Ada programmers are not so insular as to never open a file of any filetype other than "ada" once in a IDE session.

Other IDE-ish plugin projects often distribute a subset of their functionality for inclusion in the Vim runtime.  However, I don't think that there is anything in this plugin that can't be distributed.  You've already made use of the extras like Rainbow Parens conditional.

> Still I will follow established (undocumented) standards and did implement Enhancement #19 will implement Enhancement #21 as well. I do consider removing the insert key mapping as they are probably not needed. I also have to think of a solution for the documentation. I like having the documentation available in the development copy but the installed copy should come without.

I've been working to get these updated across **all** Vim filetype plugins most of which I do not and will never use.  You're the only maintainer of the maybe one hundred that I've contacted to respond with irritation let alone a "whining" post.

**I don't even use this plugin...** :wink:

**Reply by krischik** [reply]
> And yet that is exactly how it's distributed in the Vim runtime, as a file type plugin in the more general sense.

I recently noted that as well and started to merge the changes made there as well (See #24 and [feature/merge-distribution](https://github.com/krischik/vim-ada/tree/feature/merge-distribution)).

I noted that the the runtime version lacks the Ada 2012 update made by @thindil and only supports Ada 2005.

> This is irrelevant anyway, as one would assume that Ada programmers are not so insular as to never open a file of any filetype other than "ada" once in a IDE session.

I usually have multiple GVim open and make heavy use of `--servername` option to make sure files are opened in the correct GVim instance. But you have a point. Using `--servername` to mange multiple GVIm instances isn't all that common.
